### PR TITLE
javaScript-core-1-Coursework-Week2 done

### PR DESCRIPTION
**Pull request by ali1373107**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Comment by ali1373107** [comment]
@Hamisakim This is done mate.

File: exercises/C-comparison-operators/exercise.js [hunk 1]
```diff
@@ -10,11 +10,11 @@ let mentorCount = 9;
 let moreStudentsThanMentors; // finish this statement
```
Comment (Hamisakim) [inline exercises/C-comparison-operators/exercise.js:10]: missed one

File: exercises/D-logical-operators/exercise2.js [hunk 2]
```diff
@@ -5,7 +5,21 @@
...
-function isNegative() {}
+function isNegative(num) {
+  return num < 0;
+}
+
+function isBetween5and10(num) {
+  return num > 5 && num < 10;
+}
+
+function isShortName(str) {
```
Comment (Hamisakim) [inline exercises/D-logical-operators/exercise2.js:16]: have a look at this one again!  Why are we trying to split on " " ?

File: exercises/G-conditionals-2/exercise-4.js [hunk 3]
```diff
@@ -8,7 +8,12 @@
   Hint: Google how to check if a string contains a word
 */
 
-function containsCode(sentence) {}
+
+function containsCode(sentence) {
```
Comment (Hamisakim) [inline exercises/G-conditionals-2/exercise-4.js:12]: This function has a syntax error! You're returns are wrong!

File: mandatory/2-function-creation.js [hunk 4]
```diff
@@ -16,7 +18,12 @@ Write a function that:
 - numbers greater 100 must be replaced with 100
 */
 
-function formatPercentage(num) {}
+function formatPercentage(num) {
+  if (num > 100)
+    return "100%";
+  num = parseFloat(num.toFixed(2));
+  return `${num}%`;
+}
```
Comment (Hamisakim) [inline mandatory/2-function-creation.js:27]: this function needs some syntax checking

File: mandatory/3-playing-computer.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ const f2 = function (a, b) {
```
Comment (Hamisakim) [inline mandatory/3-playing-computer.js:28]: you haven't answered these questions?!

**Review by Hamisakim — COMMENTED** [review]
Good job Ali. Make sure you're testing your functions,  a few have syntax errors  and make sure to answer all questions :)

**Pull request closed without merging**